### PR TITLE
fix: search and read can now display float64 without data loss

### PR DESCRIPTION
**Pull request by adilansari**
- Read raw bytes from indexing store
- Decode data to get numerics as json.Number `json.Number`
- Validated on both read and search path:

#### Insert
```json
{
    "documents": [
        {
            "id": 9223372036854775807,
            "brand": "Nike",
            "price": 9223372036854775800,
            "review": {
                "user": "zinky",
                "votes": {
                    "upvotes": 12,
                    "downvotes": 9223372036854999999
                }
            }
        }
    ]
}
```
#### Get /id
```json
{
    "result": {
        "data": {
            "id": 9223372036854775807,
            "brand": "Nike",
            "review": {
                "votes": {
                    "downvotes": 9223372036854999999,
                    "upvotes": 12
                },
                "user": "zinky"
            },
            "price": 9223372036854775800,
        },
        "metadata": {
            "created_at": "2022-09-22T07:08:36.170413542Z"
        },
        "resume_token": "TmlrZQ=="
    }
}
```

#### Search
```json
"hits": [
            {
                "data": {
                    "review": {
                        "user": "zinky",
                        "votes": {
                            "downvotes": 9223372036854999999,
                            "upvotes": 12
                        }
                    },
                    "brand": "Nike",
                    "id": 9223372036854775807,
                    "price": 9223372036854775800
                },
                "metadata": {
                    "created_at": "2022-09-22T07:08:36.170413542Z"
                }
            }
        ],
```

### CODE REVIEW

**Review by pboros — DISMISSED** [review]

File: store/search/ts.go [hunk 1]
```diff
@@ -210,14 +214,23 @@ func (s *storeImpl) Search(_ context.Context, table string, query *qsearch.Query
 		})
 	}
 
-	res, err := s.client.MultiSearch.Perform(&tsApi.MultiSearchParams{}, tsApi.MultiSearchSearchesParameter{
+	res, err := s.client.MultiSearch.PerformWithContentType(&tsApi.MultiSearchParams{}, tsApi.MultiSearchSearchesParameter{
```
Comment (himank) [inline store/search/ts.go:217]: nice

File: store/search/ts.go [hunk 2]
```diff
@@ -210,14 +214,23 @@ func (s *storeImpl) Search(_ context.Context, table string, query *qsearch.Query
...
 	if err != nil {
 		return nil, s.convertToInternalError(err)
 	}
 
-	return res.Results, nil
+	reader := bytes.NewReader(res.Body)
+	decoder := jsoniter.NewDecoder(reader)
+	decoder.UseNumber()
+
+	var dest tsApi.MultiSearchResult
```
Comment (himank) [inline store/search/ts.go:228]: For now, we can use it but later we should just unmarshal it into structs defined in the service layer so we can optimize this path and only maintain a single copy of all these internal structs.

File: store/search/ts.go [hunk 3]
```diff
@@ -21,6 +21,8 @@ import (
 	"io"
 	"net/http"
 
+	"bytes"
```
Comment (himank) [inline store/search/ts.go:24]: go imports

File: server/search/hits.go [hunk 4]
```diff
@@ -130,8 +133,17 @@ func hitsComparator(this, that *Hit, sortingOrder *sort.Ordering) int {
 		var thisV, thatV float64
 
 		switch thisVal.(type) {
-		case float32, float64, int, int64:
-			thisV, thatV = thisVal.(float64), thatVal.(float64)
+		case json.Number:
```
Comment (himank) [inline server/search/hits.go:136]: may be add a comment here that we expect json.Number here as we are using json decoder.

File: store/search/ts.go (hunk 2 above)
Comment (adilansari) [reply store/search/ts.go:227]: Yeah makes sense

**Review by himank — APPROVED** [review]

**Pull request merged**